### PR TITLE
pkp/pkp-lib#8651 job runner enabled by default with increased job processing at request life cycle

### PR DESCRIPTION
**Pull request by touhidurabir**
This PR aims to provide solution for pkp/pkp-lib#8651

### CODE REVIEW

File: classes/core/PKPQueueProvider.php [hunk 1]
```diff
@@ -139,11 +140,21 @@ public function register()
 
         $this->registerDatabaseConnector(app()->get(\Illuminate\Queue\QueueManager::class));
 
-        if (!Application::get()->isUnderMaintenance() && Config::getVar('queues', 'job_runner', false)) {
+        if (!Application::get()->isUnderMaintenance() && Config::getVar('queues', 'job_runner', true)) {
+            
+            // This is required to make the upgrade test pass when updaring from 3.2.1 and 3.2.0
```
Comment (asmecher) [inline classes/core/PKPQueueProvider.php:145]: *upgrading

File: classes/core/PKPQueueProvider.php [hunk 2]
```diff
@@ -139,11 +140,21 @@ public function register()
...
+            // As for those version we did not have any queue related functionality 
+            // and jobs related tables are missing
+            if (!Schema::hasTable('jobs')) {
```
Comment (asmecher) [inline classes/core/PKPQueueProvider.php:148]: The job runner should not be running during maintenance due to the `isUnderMaintenance` check above; can you look into why that's not catching the upgrade?

File: xml/importexport.xsd [hunk 3]
```diff
@@ -14,7 +14,7 @@
 	<!-- Identifies a particular I18N locale, e.g. en -->
 	<simpleType name="locale">
 		<restriction base="string">
-			<pattern value="[a-z][a-z]_[A-Z][A-Z](@[a-z]+)?" />
+			<pattern value="([a-z]{2})((_[A-Z]{2})?)(@[a-z]{0,})?" />
```
Comment (asmecher) [inline xml/importexport.xsd:17]: :+1: I'll cherry-pick this in separately, thanks!

**Review by asmecher — CHANGES_REQUESTED** [review]

File: xml/importexport.xsd (hunk 3 above)
Comment (asmecher) [reply xml/importexport.xsd:17]: I extended the regex slightly to allow e.g. `ckb` (Kurdish): https://github.com/pkp/pkp-lib/commit/66f096ef37dd5b25195dfafe51257c56a6f22827

File: classes/core/PKPQueueProvider.php (hunk 2 above)
Comment (touhidurabir) [reply classes/core/PKPQueueProvider.php]: The implementation was bit misleading . we need to check the maintenance mode with the `register_shutdown_function` as service provider `register` method get called at the initial app bootstrapping time . Also moved the job runner code to `boot` method as `register` is not the right place for such implementation .

Comment (asmecher) [reply classes/core/PKPQueueProvider.php]: Thanks, @touhidurabir; is this ready for review/merging?

Comment (touhidurabir) [reply classes/core/PKPQueueProvider.php]: @asmecher Yes, it is . `OMP` test has passed but `OJS/OPS` still running. But should be ready to merge .

**Pull request merged**